main.js

```diff
@@ -1,144 +1,225 @@
+            img_element = document.createElement("img"),
+            card_body = document.createElement("div"),
+            h5 = document.createElement("h5"),
+            population = document.createElement("p"),
+            region = document.createElement("p"),
+            capital = document.createElement("p");
 
-}
+        card.classList.add("card");
+        card.style.width = "20rem";
 
-darkMode.addEventListener("click", getDarkMode)
-
-function getCountries(){
-let container = ''
-for(let i =0 ; i< countries.length ; i++){
-    container+= `
-                <div class="card" style="width: 20rem;" >
-                    <img src="${countries[i].flags.png}" class="card-img-top" alt="...">
-                    <div class="card-body default cards-styling">
-                    <h5 class="card-title">${countries[i].name}</h5>
-                    <p class="card-text"><strong>Population</strong> : ${countries[i].population}</p>
-                    <p class="card-text"><strong>Region</strong> : ${countries[i].region}</p>
-                    <p class="card-text"><strong>Capital</strong> : ${countries[i].capital}</p>
-                    </div>
-                </div>
-    `
-}
-main.innerHTML =container;
+        img_element.src = country.flags.png;
+        img_element.alt = country.name;
+        img_element.classList.add("card-img-top");
+
+        card_body.classList.add("card-body", "default", "cards-styling");
+
+        h5.classList.add("card-title");
+        h5.innerText = country.name;
+
+        population.classList.add('caard-text');
+        region.classList.add('caard-text');
+        capital.classList.add('caard-text');
+
+        population.innerHTML = `<strong>Population</strong> : ${country.population}`;
+        region.innerHTML = `<strong>Region</strong> : ${country.region}`;
+        capital.innerHTML = `<strong>Capital</strong> : ${country.capital}`;
+
+        card.appendChild(img_element);
+        card.appendChild(card_body);
+        card_body.appendChild(h5);
+        card_body.appendChild(population);
+        card_body.appendChild(region);
+        card_body.appendChild(capital);
+        // }
+        main.appendChild(card);
+    });
+    // main.innerHTML = container;
 }
-function fillOptions(){
-    let container2 = '<option value="">Filter By Region</option>'
-    let regionContainer = new Set();
-    let prevRegion = null;
-    for(let i =0 ; i < countries.length ; i++){
-            regionContainer.add(countries[i].region)
-    }
-    let regionContainerArray = Array.from(regionContainer)
-    for(let i =0 ; i<regionContainerArray.length ; i++){
-            container2 +=`
-                    <option value="${countries[i].region}">${countries[i].region}</option>
-                    `
-    }
+function fillOptions() {
+    let container2 = '<option value="" disabled selected>Filter By Region</option>';
+    let regionContainer = [];
+    // for (let i = 0; i < countries.length; i++) {

```

Comment:
201 : 203 => -1 , 
207 : 212 => -1

why 2 loops in a thing can be done in one loop ?
the loops can take from ram and CPU so we try to optimize the code. 

foreach method is more efficient.